app/src/main/java/org/kiwix/kiwixmobile/utils/StorageUtils.java

```diff
@@ -18,10 +18,29 @@
 package org.kiwix.kiwixmobile.utils;
 
 public class StorageUtils {
+  /**
+   * url parts matching this regex are removed by {@link #removeParts(String)}
+   */
+  private static final String REMOVE_URL_PART_REGEX = "\\.meta4";
 
+  /**
+   * transform an <code>url</code> to a filename by applying specific String replacement rules
+   * @param url the raw url to be adjusted
+   * @see #removeParts(String)
+   */
   public static String getFileNameFromUrl(String url) {
-    String filename = NetworkUtils.getFileNameFromUrl(url);
-    filename = filename.replace(".meta4","");
-    return filename;
+    return removeParts(NetworkUtils.getFileNameFromUrl(url));
+  }
+
+  /**
+   * remove parts matching {@link #REMOVE_URL_PART_REGEX} from given <code>filename</code>
+   * @param filename blank will return empty String
+   */
+  private static String removeParts(String filename) {
+    if (filename == null || "".equals(filename)) {
```

Comment:
You can call [isEmpty](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#isEmpty()) here. Please rebase your branch on the master and as @soloturn mentioned the logic of the single method was easier to grasp, though you keep the private constant.

Reply:
getFileNameFromUrl does not express how the method is doing its job. From the class name and fact that there is no doc why exactly and only .meta4 has to be removed, I derived potential future need for further string patterns to be removed or otherwise transformed in a second method (Seperation of Concern). In the end, this is somehow a matter of style, and the aspects mentioned become more relevant in more complex contexts.

Reply:
about the string empty check: was it ever a topic to introduce the usage of string utils library like  [StringUtils](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html) to ref its [constant empty](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#EMPTY) and method isEmpty(), or was this considered to be too heavy to bundle?

Reply:
lol, @chris1b , yes. the name storageutils itself can distract ones mind and its surprising what is in there then. but adding logic to abstract away from a name which is not clear or abstract away why only meta4 is handled this way seems to be the wrong approach. also i'd advocate to add a more complex case when a complex case is needed.

Reply:
for the empty, not sure how to put this correclty.  stringutils is 300K. i find it big just for notnull. in this case here, it cannot be null or empty:
```
    String filename;
    if (index > 1) {
      filename = url.substring(url.lastIndexOf('/') + 1, index);
    } else {
      filename = url.substring(url.lastIndexOf('/') + 1);
    }

    if ("".equals(filename.trim())) {
      filename = String.valueOf(UUID.randomUUID());
    }
```

Reply:
looking at the connection NetworkUtils / StorageUtils now, i think StorageUtils should be deleted. its method should go with a proper name into NetworkUtils. or - not merge the pull request and do nothing.

@chris1b , i am absolutely grateful that you tackle the single biggest weak spot of kiwix, which gives it a bad rating on playstore: the download of content. i love as well the way you are doing it, by changing code, and with it make people look into the source code, make them understand it better.

Reply:
So, how to proceed with StorageUtils #956?

I suggest to
1. change to StorageUtils #956 summary and description to something like 'refactor StorageUtils' respecting the note of @soloturn above (and to assign it to the person knowing about its context)
1. remove/deactivate (how?) this PR #957
1. delete feature/kiwix_cleancode_StorageUtils